miio/integrations/petfeeder/status.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict
+
+from miio import DeviceStatus
+
+
+class FoodLevel(Enum):
+    Normal = 0
+    Low = 1
+    Empty = 2
+
+
+class FoodOutletStatus(Enum):
+    Ok = 0
+    Blocked = 1
+
+
+class FoodStorageLid(Enum):
```

Comment:
Simply use a bool for boolean values, it's not worth to make it more complicated.

Reply:
Kind of agree, but also there's _a lot_ of modules doing exactly this already for bool enums.
It makes readability of the cli better.

Perhaps worth having some basic ones in a class that can be reused, much like home assistant provides?

Reply:
Having common types for these could be useful, yes, but IMO it doesn't make much sense prior having some common APIs among devices and device types (which would be the priority, having the same api among lights, vacuums, etc.).

This code base has grown organically for some years, so there are plenty of things to be found that shouldn't have been done so in the first place :-)

Reply:
From hass:
```
battery: on means low, off means normal
battery_charging: on means charging, off means not charging
cold: on means cold, off means normal
connectivity: on means connected, off means disconnected
door: on means open, off means closed
garage_door: on means open, off means closed
gas: on means gas detected, off means no gas (clear)
heat: on means hot, off means normal
light: on means light detected, off means no light
lock: on means open (unlocked), off means closed (locked)
moisture: on means moisture detected (wet), off means no moisture (dry)
motion: on means motion detected, off means no motion (clear)
moving: on means moving, off means not moving (stopped)
occupancy: on means occupied, off means not occupied (clear)
opening: on means open, off means closed
plug: on means device is plugged in, off means device is unplugged
power: on means power detected, off means no power
presence: on means home, off means away
problem: on means problem detected, off means no problem (OK)
running: on means running, off means not running
safety: on means unsafe, off means safe
smoke: on means smoke detected, off means no smoke (clear)
sound: on means sound detected, off means no sound (clear)
tamper: on means tampering detected, off means no tampering (clear)
update: on means update available, off means up-to-date
vibration: on means vibration detected, off means no vibration (clear)
window: on means open, off means closed
```
on/off, open/closed, locked/unlocked for example standout as pretty re-usable.

heh the joy of organic code :) 

For me, I'm happy to return bools - but I think there needs to be some thought about how to display these in the CLI.
Perhaps even just some state-mapping like above but only on CLI output?

Reply:
All of those are still booleans, only the presentation differs depending on what sort of device it is and homeassistant is not either consistent with this (how does a garage door differ from a regular door?).

The cli is rather something that is helpful for debugging and very simple usecases, and it was never meant to be an API itself, so for the CLI output it's fine to be rough on edges. The people who know how to install the library and use the cli tool can already interpret the output (and likely read the code if really needed).

Reply:
I gave up trying to work out how the hass team make decisions a loooong time ago 🤣 

Ok that clarifies things a little then, as I say I was basing my decision largely on other device modules providing a friendly cli output and didn't want to do less than that if you see what I mean.

FoodLevel is the only enum I'll keep then since it is an enum int and not bool.